electron/electron.js

```diff
@@ -15,16 +15,30 @@ if (squirrel) {
   app.quit();
 }
 
+// Helper function
+function downloadBounceOnMac(savePath) {
+  if (process.platform === 'darwin') {
+    app.dock.downloadFinished(path.join(savePath));
+  }
+}
+
 // declare mainWindow ahead of time so that it can be accessed in .handle('get-file')
 let mainWindow;
+// Path to Icon file
+const iconPath = path.join(__dirname, 'static/icon.png');
+// Set the icon in the Mac OS Dock
+app.dock.setIcon(iconPath);
```

Comment:
It looks like even when the app is built using `npm run build`, the icon is only set once the app is running. The stock electron icon remains in use both within Finder and for a second once the app starts or closes.

https://user-images.githubusercontent.com/53443041/140146066-ec8203a3-0ca2-4876-8bef-819e19f2d0c9.mov

I remember us running into this issue when we first tried to change the icon and I'm not quite sure what the solution is 🥴. This may be something that occurs during distribution, and if that's the case then I think we should just save this for a separate task.